benches/impact-metrics.rs

```diff
@@ -0,0 +1,65 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use std::{collections::BTreeMap, hint::black_box};
+use unleash_types::{
+    client_metrics::{ImpactMetric, ImpactMetricEnv, MetricSample, MetricType},
+    MergeMut,
+};
+
+fn generate_labels(start: usize, end: usize) -> BTreeMap<String, String> {
+    (start..end)
+        .map(|i| (format!("label_{}", i), format!("value_{}", i)))
+        .collect()
+}
+
+pub fn bench_merge(c: &mut Criterion) {
+    let first = ImpactMetricEnv {
+        app_name: "test_app".to_string(),
+        environment: "development".to_string(),
+        impact_metric: ImpactMetric {
+            name: "test_metric".to_string(),
+            help: "something".to_string(),
+            r#type: MetricType::Counter,
+            samples: vec![
+                MetricSample {
+                    value: 1.0,
+                    labels: Some(generate_labels(0, 100)),
```

Comment:
realistic case is about 5 labels: appName, env and 1-3 flags

Reply:
The data size here doesn't need to be realistic - the optimizations are linear not quadratic here. This is to get a stable benchmark test, smaller data sizes are dominated by noise